File: packages/llm-test-harness-python/llm_test_harness/metrics/regex.py
from __future__ import annotations

import re

from ..types import MetricFn, MetricScore


class Regex(MetricFn):
    def __init__(self, pattern: str | re.Pattern[str], flags: int = 0) -> None:
        if isinstance(pattern, re.Pattern):
            self._pattern = pattern
        else:
            self._pattern = re.compile(pattern, flags)

    @property
    def name(self) -> str:
        return "Regex"

    def evaluate(self, text: str) -> MetricScore:
        passed = bool(self._pattern.search(text))
        return MetricScore(
            name=self.name,
            passed=passed,
            score=1.0 if passed else 0.0,
        )

File: packages/llm-test-harness/src/metrics/regex.ts
import type { MetricFn, MetricScore } from '../types.js'

export class Regex implements MetricFn {
  readonly name = 'Regex'
  private pattern: RegExp

  constructor(pattern: RegExp | string, flags?: string) {
    this.pattern = pattern instanceof RegExp ? pattern : new RegExp(pattern, flags)
  }

  evaluate(text: string): MetricScore {
    const pass = this.pattern.test(text)
    return { name: this.name, pass, score: pass ? 1 : 0 }
  }
}
